projects/mysadna/src/app/components/project-details/project-details.component.ts

```diff
@@ -12,36 +12,28 @@ import { FirebaseService } from '@/services';
 })
 export class ProjectDetailsComponent implements OnInit {
   project: Project;
-
   projectUsers: User[];
 
   constructor(
     private route: ActivatedRoute,
-    private firebaseService: FirebaseService,
-    private location: Location
+    private firebaseService: FirebaseService
   ) { }
 
-  ngOnInit() {
-    this.getProject();
-  }
-
-  getProject(): void {
+  ngOnInit(): void {
     const projectId: number = +this.route.snapshot.paramMap.get('projectId');
-    this.firebaseService.getProjectList()
-      .subscribe(projectList => {
-        this.project = projectList.find(project => (project.getProjectId() === projectId));
-        this.getProjectUsers(projectId);
-      });
-  }
-
-  getProjectUsers(projectId: number): void {
-    this.firebaseService.getUserList()
-      .subscribe(users => {
-        this.projectUsers = users.filter(user => user.getProjectList().includes(projectId));
-      });
+    this.loadData(projectId);
   }
 
-  goBack(): void {
-    this.location.back();
+  loadData(projectId: number): void {
+    zip(
+      this.firebaseService.getUserList(),
+      this.firebaseService.getProjectList()
+    ).subscribe(data => {
```

Comment:
Here's we can use the sync way, which we discussed.
To not create queue, we can load users and projects at the same time, and subscribe on changes in one place.

Reply:
Does `zip` create an array the length of which equals the number of calls made within the `zip` and then insert the return value of each call into the corresponding index?

Reply:
@a3l5o7n9 Exactly!